### PR TITLE
chore: move wmg api endpoints to separate api spec file

### PR DESCRIPTION
**Pull request by atolopko-czi**
### Reviewers
**Functional:** 
@Bento007 

**Readability:** 
@MDunitz 

---

## Changes
- move wmg endpoints to wmg-api.config file, out of corpora-api.config
- upgrade connexion to latest version & update connexion usage as needed for multi-api support
- add in top-level index.html page for navigating to each of the independent api's swagger UI pages
- move data portal API's swagger UI to /dp base path; this also allows us to remove /dp prefix from each endpoint in the spec file.

### CODE REVIEW

File: tests/unit/backend/corpora/api_server/test_revisions.py [hunk 1]
```diff
@@ -411,10 +411,10 @@ def publish_collection(self, collection_id: str) -> dict:
 
         return response_json
 
-    def verify_datasets(self, actual_body: dict, expected_dataset_ids: typing.List[str]) -> None:
+    def verify_datasets(self, actual_body: dict, expected_dataset_ids: typing.Set[str]) -> None:
```
Comment (atolopko-czi) [inline tests/unit/backend/corpora/api_server/test_revisions.py:414]: unrelated fix for transient test failures

**Review by MDunitz — APPROVED** [review]
lgtm

File: backend/config/wmg-api.yml [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  - description: Local
+    url: /
+  - description: Production environment
+    url: https://api.cellxgene.cziscience.com/
+  - description: Development environment
+    url: https://api.dev.single-cell.czi.technology/
+  - description: Staging environment
+    url: https://api.staging.single-cell.czi.technology/
+paths:
+  /wmg/v1/primary_filter_dimensions:
```
Comment (Bento007) [inline backend/config/wmg-api.yml:17]: remove wmg
```suggestion
  /v1/primary_filter_dimensions:
```

File: backend/config/wmg-api.yml [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+                    $ref: "#/components/schemas/wmg_ontology_term_id_label_list"
+                  tissue_terms:
+                    $ref: "#/components/schemas/wmg_ontology_term_id_label_list"
+                  gene_terms:
+                    # TODO: specify further: https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/1967
+                    # `[{"<ontology_term_id>": "<ontology_term_label"}, ...]`
+                    type: object
+        "404":
+          $ref: "#/components/responses/404"
+  /wmg/v1/query:
```
Comment (Bento007) [inline backend/config/wmg-api.yml:45]:
```suggestion
  /v1/query:
```

File: tests/unit/backend/corpora/api_server/base_api_test.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 class BaseAPITest(DataPortalTestCase):
     """
-    Provide access to the test Corpora API. All test for the Corpora API should inherit this class.
+    Provide access to the test APIs. All tests for APIs should inherit this class.
```
Comment (Bento007) [inline tests/unit/backend/corpora/api_server/base_api_test.py:14]: :+1:

**Review by Bento007 — CHANGES_REQUESTED** [review]
missed some stuff

**Pull request merged**